packages/flutter/lib/src/cupertino/theme.dart

```diff
@@ -24,38 +24,69 @@ const Color _kDefaultBarDarkBackgroundColor = Color(0xB7212121);
 /// [CupertinoTheme.of]. An [InheritedWidget] dependency is created when
 /// an ancestor [CupertinoThemeData] is retrieved via [CupertinoTheme.of].
 ///
+/// The [CupertinoTheme] widget implies an [IconTheme] widget, whose
+/// [IconTheme.data] has the same color as [primaryColor]
+///
 /// See also:
 ///
 ///  * [CupertinoThemeData], specifies the theme's visual styling.
 ///  * [CupertinoApp], which will automatically add a [CupertinoTheme].
 ///  * [Theme], a Material theme which will automatically add a [CupertinoTheme]
 ///    with a [CupertinoThemeData] derived from the Material [ThemeData].
-class CupertinoTheme extends InheritedWidget {
+class CupertinoTheme extends StatelessWidget {
   /// Creates a [CupertinoTheme] to change descendant Cupertino widgets' styling.
   ///
   /// The [data] and [child] parameters must not be null.
   const CupertinoTheme({
     Key key,
     @required this.data,
-    @required Widget child,
+    @required this.child,
   }) : assert(child != null),
        assert(data != null),
-       super(key: key, child: child);
+       super(key: key);
 
   /// The [CupertinoThemeData] styling for this theme.
   final CupertinoThemeData data;
 
-  @override
-  bool updateShouldNotify(CupertinoTheme oldWidget) => data != oldWidget.data;
-
   /// Retrieve the [CupertinoThemeData] from an ancestor [CupertinoTheme] widget.
   ///
   /// Returns a default [CupertinoThemeData] if no [CupertinoTheme] widgets
   /// exist in the ancestry tree.
   static CupertinoThemeData of(BuildContext context) {
-    final CupertinoTheme theme = context.inheritFromWidgetOfExactType(CupertinoTheme);
-    return theme?.data ?? const CupertinoThemeData();
+    final _InheritedCupertinoTheme inheritedTheme = context.inheritFromWidgetOfExactType(_InheritedCupertinoTheme);
+    return inheritedTheme?.theme?.data ?? const CupertinoThemeData();
   }
+
+  /// The widget below this widget in the tree.
+  ///
+  /// {@macro flutter.widgets.child}
+  final Widget child;
+
+  @override
+  Widget build(BuildContext context) {
+    final IconThemeData iconTheme = IconTheme.of(context) ?? const IconThemeData();
```

Comment:
Seems like Material doesn't bring anything from the tree above it. I think it makes sense here too where if you insert a *App into the tree, everything's fresh.